### PR TITLE
[REVIEW] Add RMM Python docs

### LINKED ISSUE #630: [DOC] Add Python docs
Currently there is no API reference or published docs for the Python API and it would be nice to have those things.

### PR DESCRIPTION
**Pull request by shwina**
Closes #630

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/docs/basics.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+```python
+>>> import rmm
+>>> pool = rmm.mr.PoolMemoryResource(
+...     upstream=rmm.mr.CudaMemoryResource(),
+...     initial_pool_size=2**30,
+...     maximum_pool_size=2**32
+... )
+>>> rmm.mr.set_current_device_resource(pool)
+```
```
Comment (kkraus14) [inline python/docs/basics.md:93]: It may be good to show an example of creating a Pool of managed memory as well.

File: python/docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+author = "NVIDIA"
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = "0.17"
+# The full version, including alpha/beta/rc tags.
+release = "0.15.0"
```
Comment (kkraus14) [inline python/docs/conf.py:74]: This looks wrong?

File: python/docs/make.bat [hunk 3]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (kkraus14) [inline python/docs/make.bat:1]: RMM doesn't support Windows, do we need to support building docs on Windows?

**Review by kkraus14 — CHANGES_REQUESTED** [review]

File: python/docs/conf.py (hunk 2 above)
Comment (kkraus14) [reply python/docs/conf.py:74]: Would also be good to capture these in https://github.com/rapidsai/rmm/blob/branch-0.17/ci/release/update-version.sh

File: python/docs/make.bat (hunk 3 above)
Comment (shwina) [reply python/docs/make.bat]: rm'd

File: python/docs/conf.py (hunk 2 above)
Comment (shwina) [reply python/docs/conf.py]: Fixed & done

File: python/docs/basics.md (hunk 1 above)
Comment (shwina) [reply python/docs/basics.md:93]: writ

**Review by kkraus14 — APPROVED** [review]

File: python/docs/basics.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
+# RMM - the RAPIDS Memory Manager
```
Comment (harrism) [inline python/docs/basics.md:1]: My concern hereis that this duplicates the content in the readme, so we have two places to edit this content. Perhaps just change the readme to a summary and link to this doc.

**Review by harrism — CHANGES_REQUESTED** [review]

File: python/docs/basics.md (hunk 4 above)
Comment (shwina) [reply python/docs/basics.md:1]: Agree that we shouldn't duplicate. Will remove from README after we get the docs published on rapids.ai

**Comment by shwina** [comment]
rerun tests

**Review by ajschmidt8 — APPROVED** [review]

File: python/docs/basics.md [hunk 5]
```diff
@@ -1,5 +1,20 @@
 # RMM - the RAPIDS Memory Manager
 
+RMM is a package that enables you to allocate GPU ("device") memory
```
Comment (harrism) [inline python/docs/basics.md:3]: It's important to say something about performance, I think.

Comment (shwina) [reply python/docs/basics.md]: How's this?

**Review by harrism — APPROVED** [review]

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**